packages/kotti-ui/source/kotti-field-file-upload/KtFieldFileUploadRemote.vue

```diff
@@ -147,6 +151,24 @@ export default defineComponent({
 				newUploadedFiles.add(payload.id)
 		}
 
+		const setStatuses = (
+			payload: Map<string, KottiFieldFileUploadRemote.Status>,
+		) => {
+			field.setValue(
+				field.currentValue.map((fileItem) => {
+					const status =
+						typeof fileItem.id === 'string' && payload.get(fileItem.id)
```

Comment:
Ah, good point... the problem here was that payload `Kotti.FieldFileUploadRemote.Payload` is typed as `Record<string, _>`, i.e. the strings are assumed to be strings. I took it to mean that the payload would always ever contain ids that we randomly generate (which I think is always a string) and therefore thought this is reasonable.

If I remove this check, there is a typescript complaint like this:
```ts
const status = payload.get(fileItem.id)
                        // ^^^^^^^^^^^
                        // Argument of type 'string | number' is not assignable to parameter of type 'string'.
```

The other way to solve this would be to cast:
```ts
const status = payload.get(String(fileItem.id))
```
I think that is the proper way to handle this.